### ISSUE TITLE
Diagnostics messages changed with no update to atom-ide-diagnostics

### ISSUE THREAD

**Issue by Arcanemagus** [facebook-atom/atom-ide-ui#103]
### Description

It appears that the structure of the `diagnostics` messages coming from `atom-ide-diagnostics` has changed with no corresponding update to the version of the service published by `atom-ide-diagnostics`.

### Expected Behavior

API behavior should stay the same unless a version change is made.

### Actual Behavior

In `atom-ide-ui@0.3.3` messages published by `atom-ide-diagnostics` had this structure:
```json
{
 "scope": "file",
 "providerName": "ESLint",
 "type": "Error",
 "filePath": "C:\\Users\\abneyl\\Documents\\GitHub\\atom-minimap-linter\\lib\\helpers.js",
 "text": "Trailing spaces not allowed. (no-trailing-spaces)",
 "range": {
  "start": {
   "row": 24,
   "column": 24
  },
  "end": {
   "row": 24,
   "column": 26
  }
 },
 "fix": {
  "oldRange": {
   "start": {
    "row": 24,
    "column": 24
   },
   "end": {
    "row": 24,
    "column": 26
   }
  },
  "newText": ""
 }
}
```

`atom-ide-ui@0.4.0` through `atom-ide-ui@0.5.1` had the same structure.

`atom-ide-ui@0.5.2` changed to the following structure:
```json
{
 "providerName": "ESLint",
 "type": "Error",
 "filePath": "C:\\Users\\abneyl\\Documents\\GitHub\\atom-minimap-linter\\lib\\helpers.js",
 "text": "Trailing spaces not allowed. (no-trailing-spaces)",
 "range": {
  "start": {
   "row": 24,
   "column": 24
  },
  "end": {
   "row": 24,
   "column": 26
  }
 },
 "fix": {
  "oldRange": {
   "start": {
    "row": 24,
    "column": 24
   },
   "end": {
    "row": 24,
    "column": 26
   }
  },
  "newText": ""
 },
 "actions": []
}
```

Note the lack of the `scope` attribute which was used in https://github.com/AtomLinter/atom-minimap-linter/pull/23 to add support for `atom-ide-ui`. (All messages now show as invalid for the current editor, causing https://github.com/AtomLinter/atom-minimap-linter/issues/29.)

I'll be updating `atom-minimap-linter` to accommodate the new message structure, but in the future that service needs to be update to note the changes when attributes are removed.

### Versions
* Atom: v1.22.0-beta1
* Client OS: Windows 10 x64
* atom-ide-ui: v0.5.2

**Comment by hansonw** [comment]
Ah, my bad! Those messages aren't totally meant to be consumed externally so I forgot that it was being used by the linter minimap. Sorry! 🙈

**Comment by hansonw** [comment]
Will definitely keep this in mind as we likely will try to deprecate that particular message API format to use something more consistent with the other Linter messages internally as well.

**Comment by Arcanemagus** [comment]
Removing checking that actually allows me to use the exact same code for `linter` and `atom-ide-ui`, I filed the issue to bring it to your attention though 😉.

**Comment by Arcanemagus** [comment]
Fix published in `minimap-linter@2.1.2` in case anyone comes here from searching 😉.
